community/kernel/src/main/java/org/neo4j/kernel/impl/util/NodeProxyWrappingNodeValue.java

```diff
@@ -103,7 +105,7 @@ public MapValue properties()
         MapValue m = properties;
         if ( m == null )
         {
-            synchronized ( this )
+            synchronized ( propLock )
```

Comment:
I remember us talking about decoupling synchronization on this class, but when I tried to completely remove synchronization it still didn't solve the regression according to the benchmarks I ran, way back i september or something. Back then I probably messed something up, but I would love to see some benchmark result to confirm this fix!

Reply:
No this should be reverted, forgot about it. I reverted it in `RelationshipProxyWrappingValue`
but forgot about this one